### DISCUSSION TITLE
Question: It's possible to add custom panel to theia debug extension?

### DISCUSSION THREAD

**Discussion by eli88p** [eclipse-theia/theia#9916] [Q&A]
Hello,
There is some way to add panel to @theia/debug widget (like in my image)

Thanks in advance.
![Capture](https://user-images.githubusercontent.com/11608934/75960887-989bb480-5ec9-11ea-8405-9e70427df037.PNG)

**Comment by akosyakov** [comment]
Yes, you should be able to grab debug view container and call `addWidget` on it via custom Theia extension. With a vscode extension you can declare that your view should belong to `debug` view container.

That's how it is implemented in the plugin system: https://github.com/eclipse-theia/theia/blob/3d5e4f98d2bc6a370b8db726568e6feecb93d20a/packages/plugin-ext/src/main/browser/view/plugin-view-registry.ts#L124-L127

Basically whenever the debug view is created it injects a new view into it. The same done on the startup in the case if the debug widget was already created: 

https://github.com/eclipse-theia/theia/blob/3d5e4f98d2bc6a370b8db726568e6feecb93d20a/packages/plugin-ext/src/main/browser/view/plugin-view-registry.ts#L447-L451
